avalon/tools/workfiles/app.py

```diff
@@ -183,173 +235,339 @@ def refresh(self):
-        separator.setFrameShadow(QtWidgets.QFrame.Sunken)
-        self.layout.addWidget(separator)
+        # Force focus on the open button by default, required for Houdini.
+        widgets["fileOpen"].setFocus()
 
-        current_file = self.host.current_file()
-        if current_file:
-            current_label = os.path.basename(current_file)
-        else:
-            current_label = "<unsaved>"
-        current_file_label = QtWidgets.QLabel("Current File: " + current_label)
-        self.layout.addWidget(current_file_label)
+        # Connect signals
+        widgets["assets"].current_changed.connect(self.on_asset_changed)
 
-        buttons_layout = QtWidgets.QHBoxLayout()
-        self.save_as_button = QtWidgets.QPushButton("Save As")
-        buttons_layout.addWidget(self.save_as_button)
-        self.layout.addLayout(buttons_layout)
+        self.widgets = widgets
 
-        self.duplicate_button.pressed.connect(self.on_duplicate_pressed)
-        self.open_button.pressed.connect(self.on_open_pressed)
-        self.list.doubleClicked.connect(self.on_open_pressed)
-        self.browse_button.pressed.connect(self.on_browse_pressed)
-        self.save_as_button.pressed.connect(self.on_save_as_pressed)
+        self.refresh()
+        self.resize(750, 500)
 
-        self.open_button.setFocus()
+    def set_context(self, context):
 
-        self.refresh()
-        self.resize(400, 550)
+        if "asset" in context:
+            asset = context["asset"]
+            asset_document = io.find_one({"name": asset,
+                                          "type": "asset"})
 
-    def get_name(self):
-        window = NameWindow(self.root)
-        window.setStyleSheet(style.load_stylesheet())
+            # Set silo
+            silo = asset_document["data"].get("silo")
+            if self.widgets["assets"].get_current_silo() != silo:
+                self.widgets["assets"].set_silo(silo)
+
+            # Select the asset
+            self.widgets["assets"].select_assets([asset], expand=True)
+
+        if "task" in context:
+            self.widgets["tasks"].select_task(context["task"])
+
+    def get_filename(self):
+        """Show save dialog to define filename for save or duplicate
+
+        Returns:
+            str: The filename to create.
+
+        """
+        window = NameWindow(parent=self,
+                            root=self.root)
         window.exec_()
 
         return window.get_result()
 
     def refresh(self):
-        self.list.clear()
+
+        self.root = self.host.work_root()
+
+        # Refresh asset widget
+        self.widgets["assets"].refresh()
+
+        # Refresh current scene label
+        current = os.path.basename(self.host.current_file()) or "<unsaved>"
+        self.widgets["fileCurrent"].setText("Current File: %s" % current)
+
+        # Refresh breadcrumbs
+        self.widgets["header"].refresh()
+
+        # Refresh files list
+        list = self.widgets["fileList"]
```

Comment:
Wouldn't it a bit danger that overriding the class `list` ? Since it's often used :o

Reply:
Well spotted, I guess I went a bit to quickly with my refactoring - will fix.